code/controllers/subsystem/air.dm

```diff
@@ -68,6 +72,15 @@ SUBSYSTEM_DEF(air)
 /datum/controller/subsystem/air/fire(resumed = 0)
 	var/timer = TICK_USAGE_REAL
 
+	if(pipenets_needing_rebuilt.len) // always do this nonsense first
+		var/oldpart = currentpart
+		rebuild_pipenets(resumed)
```

Comment:
I would unwrap this proc, it's called a ton

Reply:
what do you mean

Reply:
Procs in byond have stupid amounts of overhead. If you're calling a proc really frequently or in a really performance critical situation where doing things *as fast as possible* such as here is a priority taking the contents of a proc and inlining them straight into the code actually has pretty significant performance improvements.
You could turn them into defines as well if you still want to preserve the code organization, but considering they're only used in one place it wouldn't be that meaningful of a difference imo.

Reply:
aight